### PR TITLE
logs: log to stdout/stderr when subp fails

### PR DESCRIPTION
**Pull request by CalvoM**
## Proposed Commit Message

logs: log to stdout/stderr when subp fails

## Desired commit type

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist

 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

**Comment by CalvoM** [comment]
@lucasmoura will this PR change later after #1905 is approved? i.e. replacing the print messages with event.info

File: uaclient/util.py [hunk 1]
```diff
@@ -676,10 +676,12 @@ def subp(
         except ProcessExecutionError as e:
             if capture:
                 logging.debug(str(e))
+                print(str(e))
             if not retry_sleeps:
                 raise
             retry_msg = " Retrying %d more times." % len(retry_sleeps)
             logging.debug(redact_sensitive_logs(str(e) + retry_msg))
+            print(str(e))
```
Comment (lucasmoura) [inline uaclient/util.py:684]: Should we only print that message if `capture` is `True` ? Otherwise we could be printing this message twice to the user

**Review by lucasmoura — COMMENTED** [review]
@CalvoM Let's also add an unittest for this change.

And regarding the `event.info` PR, no worries at all. If this PR gets merged first, I can do the change on my side.
And even though I think your PR will be merged first, if that does not happen, then yes, we will need to change this PR, but it will be a simple change

**Review by orndorffgrant — CHANGES_REQUESTED** [review]
Hi @lucasmoura and @CalvoM Sorry I think I miscommunicated the intent of this feature in the issue description.

My request was not to log the current error to stdout.

My request was to take the stderr/stdout of the subprocess and make sure we are always logging that when the subprocess fails. (I've added the mattermost conversation that triggered this issue to the issue description).

Though now I see we have the `capture` argument to `subp`. So I think we should only log stdout/stderr of the subprocess when `capture=True`

Altogether, I think the change would involve adding some code after `util.py:678` that will `log.debug` `e.stdout` and `e.stderr` maybe with some nice string formatting to add labels.

What do you think @lucasmoura @CalvoM ?

**Comment by lucasmoura** [comment]
@orndorffgrant That makes sense to me and I agree that is indeed a better approach

**Comment by CalvoM** [comment]
@orndorffgrant, with the context you have provided, it is surely a better approach.
@lucasmoura  and @orndorffgrant This message from [run_command](https://github.com/canonical/ubuntu-advantage-client/blob/c89d1cab1de5c055167ce23c5193f84f89119d86/lib/reboot_cmds.py#L32) in reboot_cmd is a good message to use. What do you think?

File: uaclient/tests/test_util.py [hunk 2]
```diff
@@ -350,6 +350,43 @@ def test_retry_logs_remaining_retries(self, m_sleep, m_subp, caplog_text):
         for log in expected_logs:
             assert log in logs
 
+    @pytest.mark.parametrize("caplog_text", [logging.WARNING], indirect=True)
+    @mock.patch("uaclient.util._subp")
+    def test_log_subp_fails_stdout_stderr_capture_true(
```
Comment (lucasmoura) [inline uaclient/tests/test_util.py:355]: Maybe we can combine both of these tests and just parametrize the `capture` parameter

Comment (CalvoM) [reply uaclient/tests/test_util.py:355]: Yes, I was reading more on pytest just about that.

Comment (CalvoM) [reply uaclient/tests/test_util.py]: Updated.

**Review by lucasmoura — APPROVED** [review]

**Review by orndorffgrant — APPROVED** [review]
LGTM! Merging

**Pull request merged**